src/game_camera.cpp

```diff
@@ -0,0 +1,302 @@
+
+#include <stdlib.h>
+#include <stdio.h>
+
+extern "C" {
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+}
+
+#include "core/system.h"
+#include "core/obb.h"
+#include "render/camera.h"
+#include "controls.h"
+#include "room.h"
+#include "world.h"
+#include "skeletal_model.h"
+#include "entity.h"
+//#include "script.h"
```

Comment:
I left those three as I though they might be used in conjunction with the current code to implement correct camera controls in the future. I can remove them if it's bad practice.